components/Elevation/src/UIView+MaterialElevationResponding.m

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIView+MaterialElevationResponding.h"
+
+#import "MDCElevation.h"
+
+@implementation UIView (MaterialElevationResponding)
+
+- (CGFloat)mdc_baseElevation {
+  CGFloat totalElevation = 0;
+
+  if (!self.superview) {
+    return totalElevation;
+  }
+  UIView *superView = self.superview;
+
+  while (superView != nil) {
+    id<MDCElevation> elevatableCurrent = [superView conformingObjectInResponderChain];
+    if (elevatableCurrent) {
+      if ([elevatableCurrent respondsToSelector:@selector(mdc_overrideBaseElevation)]) {
+        totalElevation += elevatableCurrent.mdc_overrideBaseElevation;
```

Comment:
From the documentation, https://github.com/material-components/material-components-ios/blob/develop/components/Elevation/src/MDCElevation.h#L42 `mdc_overrideBaseElevation` shouldn't be used in `mdc_baseElevation`?  Maybe clarify the documentation there to be `mdc_currentElevation` or `zero` if I understand correctly? I feel `zero` is more proper?

Reply:
`mdc_overrideBaseElevation` is there to override whatever the `baseElevation` is so that clients can if the want to _correct_ the elevation. I'm confused by the comment regarding `mdc_current` and `zero`.

Reply:
Umm..`mdc_overrideBaseElevation` doesn't look like an `override` value from the code, it looks more like an `additionalElevation`.  Because `mdc_baseElevation` is a sum of all superview's values. and `mdc_overrideBaseElevation`  adds an additional value into it, it seems like it doesn't override anything.

Reply:
If there is an override then base returns the current sum plus the `override`, if a view responds to `override` then its `base` would simply be `override`.